### PR TITLE
Ankit | Test  | "Connect to HMRC" button should come on VAT payments/liabilities and obligation page if any error comes from API 

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cwgvbru

### CODE REVIEW

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.html [hunk 1]
```diff
@@ -98,8 +98,18 @@
                         <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
                     </table>
                     <section *ngIf="!tableDataSource?.length">
-                        <div class="no-data">
-                            <h1>{{ commonLocaleData?.app_no_data_found }}</h1>
+                        <div *ngIf="getFormControl('taxNumber').value">
+                            <div *ngIf="connectToHMRCUrl" class="initial-screen position-absolute text-center">
+                                <a mat-stroked-button color="primary" class="text-center" [href]="connectToHMRCUrl">
```
Comment (nisha-walkover) [inline apps/web-giddh/src/app/vat-report/obligations/obligations.component.html]: use button tag

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.html [hunk 2]
```diff
@@ -98,8 +98,18 @@
                         <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
                     </table>
                     <section *ngIf="!tableDataSource?.length">
-                        <div class="no-data">
-                            <h1>{{ commonLocaleData?.app_no_data_found }}</h1>
+                        <div *ngIf="getFormControl('taxNumber').value">
+                            <div *ngIf="connectToHMRCUrl" class="initial-screen position-absolute text-center">
```
Comment (nisha-walkover) [inline apps/web-giddh/src/app/vat-report/obligations/obligations.component.html]: why using position absolute ?

**Review by nisha-walkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.scss [hunk 3]
```diff
@@ -16,4 +16,11 @@
     .action-btn {
         min-width: 125px;
     }
+        
+    .initial-screen {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/obligations/obligations.component.scss]: .position-center

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts [hunk 4]
```diff
@@ -126,7 +135,42 @@ export class ObligationsComponent implements OnInit, OnDestroy {
...
+    * Get Current company branches information
+    *
+    * @private
+    * @memberof ObligationsComponent
+    */
+    private getCurrentCompanyBranchTaxNumber(): void {
+        this.componentStore.currentCompanyBranches$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                const currentBranchUniqueName = this.generalService.currentBranchUniqueName;
+                let currentBranch = cloneDeep(response.find(branch => branch?.uniqueName === currentBranchUniqueName));
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts]: check without cloneDeep if work remove it

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.html (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.html]: Check use of text center if not required remove it.

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.scss [hunk 5]
```diff
@@ -1,4 +1,11 @@
 .liabilities-payments {
+    min-height: calc(100vh - 90px);
+    position: relative;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.scss]: use class "position-relative"

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts [hunk 6]
```diff
@@ -336,4 +342,16 @@ export class VatLiabilitiesPayments implements OnInit, OnDestroy {
...
     }
+    /**
+     * This will call API to get HMRC get authorization url
+     *
+     * @memberof VatLiabilitiesPayments
+     */
+    public getURLHMRCAuthorization(): void {
+        this.vatService.getHMRCAuthorization(this.activeCompany.uniqueName).pipe(takeUntil(this.destroyed$)).subscribe((res) => {
+            if (res?.body) {
+                this.connectToHMRCUrl = res?.body;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts]:
```suggestion
                this.connectToHMRCUrl = res.body;
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.html (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.html:103]: Done

Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.html:103]: connectToHMRCUrl has unique url , not redirect in giddh website.

Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.html]: connectToHMRCUrl has unique url , not redirect in giddh website.

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.html (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.html]: remove text-center DONE.
{{ localeData?.connect_to_hmrc }} in center  ,  use of position absolute.

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.scss (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.scss]: Done

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.scss (hunk 5 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.scss]: Done

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts (hunk 6 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.html [hunk 7]
```diff
@@ -40,12 +40,12 @@
                 </div>
             </div>
 
-            <div class="mt-5" *ngIf="isLoading">
-                <giddh-page-loader></giddh-page-loader>
+            <div class="position-center position-absolute" *ngIf="isLoading">
+                <giddh-page-loader class="unset-loader-margin"></giddh-page-loader>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/obligations/obligations.component.html:44]: add this class declaration in global style

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.scss [hunk 8]
```diff
@@ -16,4 +16,11 @@
     .action-btn {
         min-width: 125px;
     }
+    min-height: calc(100vh - 90px);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/obligations/obligations.component.scss]: move in line no 2

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts [hunk 9]
```diff
@@ -126,34 +137,75 @@ export class ObligationsComponent implements OnInit, OnDestroy {
...
-            if (response?.status === "success" && response?.body?.obligations) {
+        this.connectToHMRCUrl$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.body) {
+                this.connectToHMRCUrl = response.body;
+            }
+        });
+
+        this.obligationList$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.body?.obligations) {
                 this.tableDataSource = response?.body?.obligations.map(item => {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts]:
```suggestion
                this.tableDataSource = response.body.obligations.map(item => {
```

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.scss [hunk 10]
```diff
@@ -1,4 +1,10 @@
 .liabilities-payments {
+    min-height: calc(100vh - 90px);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.scss:2]:
```suggestion
    min-height: calc(100vh - 90px);
    
```

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts [hunk 11]
```diff
@@ -93,6 +97,8 @@ export class VatLiabilitiesPayments implements OnInit, OnDestroy {
             if (activeCompany) {
                 this.activeCompany = activeCompany;
                 this.getFormControl('companyUniqueName').patchValue(activeCompany.uniqueName);
+                this.getURLHMRCAuthorization();
+
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts]:
```suggestion
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.html (hunk 7 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.html:44]: Done

Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.html:44]: Already exists.

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.scss (hunk 8 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.scss]: Done

File: apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts (hunk 9 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.scss (hunk 10 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.scss:2]: Done

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts (hunk 11 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts]: Done

**Review by dvCodeWorld — APPROVED** [review]

**Review by nisha-walkover — APPROVED** [review]

**Pull request merged**